pkg/mover/chart.go

```diff
@@ -68,34 +68,51 @@ func (l *defaultLogger) Println(i ...interface{}) {
 	fmt.Println(i...)
 }
 
+// ChartMetadata exposes metadata about the Helm Chart to be relocated
+type ChartMetadata struct {
+	Name    string
+	Version string
+}
+
+// RegistryCredentials to access a registry
+type RegistryCredentials struct {
+	Username, Password string
+	// add other credentials field/options as needed
+}
+
+// ChartMoveRequest defines a chart move
+type ChartMoveRequest struct {
+	Chart          string
+	ImageHintsFile string
+	Rules          RewriteRules
+	FromCreds      RegistryCredentials
```

Comment:
I'd like to step back for a second

Eventually we will have potentially two new requirements

* Source/target repository definitions for Helm Chart repositories (not for container images)
* Multi-step relocation process for airgap https://github.com/vmware-tanzu/asset-relocation-tool-for-kubernetes/issues/15

Taking that into account I'd say that the proposed `ChartMoveRequest` struct is too flat to enable this extensibility in the future.

I'd suggest to take a look at how chart syncer [implements its API](https://github.com/bitnami-labs/charts-syncer/blob/master/api/config.proto) and we think about something similar.

To me, these are the nobs that we should be able to define for a relocation

* Source
  * Chart (currently only local path) // eventually maybe an intermediate imgpkg bundle
  * Container images (Currently only OCI) // eventually maybe an intermediate imgpkg bundle
* Target   
  * Chart (currently only local path)
  * Container images (Currently only OCI)
* ChartRewrites


The scope of this task is to extend source.containerImages and target.ContainterImages but in such a way we can leave room to the upcoming new requirements.

WDYT?

Reply:
I can add the Source and Target, for now it just seemed too much for what we had.

Also we might need to end up using something to help us construct the request struct if it gets too complex... hopefully now.

BTW the credentials are not in your example structure above, but I would assume they are at the top level, as they can be for the source or target registries in-distinctively, as @petewall already pointed out.

Reply:
BTW why are not the rewrites part of the target section?

Reply:
> I can add the Source and Target, for now it just seemed too much for what we had.
Not sure I understand, is it or is it not too much?

> Also we might need to end up using something to help us construct the request struct if it gets too complex... hopefully now.

Could you elaborate? To me, that struct content will mostly get eventually marshaled from a file.

> BTW the credentials are not in your example structure above, but I would assume they are at the top level, as they can be for the source or target registries in-distinctively, as @petewall already pointed out.

Yes, those are just top levels but notice how I explicitly called out two different kind of targets and sources to leave room to add the capability of pulling charts, pulling intermediate bundles and pushing charts